### ISSUE TITLE
Missing toolchain leads to error in cargo call (on the example of building rom_ext_signer)

### ISSUE THREAD

**Issue by e7p** [lowRISC/opentitan#7407]
When trying to build normally using `ninja -C build-out all`, I get the following error:

```
[9/2194] Generating rom_ext_signer with a custom command
error: no such subcommand: `+`

	Did you mean `b`?
[19/2194] Generating 'sw/device/lib/dif/5fb29ed@@uart_ot@sta/uart_regs.h'.
FAILED: sw/host/rom_ext_image_tools/signer 
/usr/bin/env 'CC=ccache cc' 'CXX=ccache c++' /home/ep/EMSEC/opentitan/opentitan/util/invoke_cargo.sh /usr/bin/cargo '--manifest-path=/home/ep/EMSEC/opentitan/opentitan/sw/host/rom_ext_image_tools/signer/Cargo.toml --target-dir=/home/ep/EMSEC/opentitan/opentitan/build-out/sw/host/rom_ext_image_tools/signer --release ' '' '' /home/ep/EMSEC/opentitan/opentitan /home/ep/EMSEC/opentitan/opentitan/build-out
[29/2194] Compiling C++ object 'sw/dev...gletest_googletest_src_gtest-all.cc.o'
ninja: build stopped: subcommand failed.
```

The problem here is that there is no toolchain file defined for the rom_ext_signer target [in this line](https://github.com/lowRISC/opentitan/blob/b3bc45b4a805b4e8d9fcca3c85e3d793ded19ffc/sw/host/rom_ext_image_tools/signer/meson.build#L52), as a previously empty one was already removed in 405dd3e (reverting this would lead to the exactly same behavior, as the toolchain is still empty).

I am aware that this was also discussed in #6654. However the proposed solution was to look for the compatible versions of the requirements cmake, rust, go, perl and meson (<= 0.54). For me, unfortunately this did not do the trick and the error is persisting.

I could come around it by modifying the [invoke_cargo.sh](https://github.com/lowRISC/opentitan/blob/93536cde3ccca091fac73a0304db54aea4ad32af/util/invoke_cargo.sh#L29) script, by removing the `+"${TOOLCHAIN}"` expression in case the toolchain is not given or empty, however I don't know if this would be the correct approach.

My system is Arch Linux based and except for meson, it is updated to most current versions.

**Comment by imphil** [comment]
Can you attach the whole build log, please?

**Comment by e7p** [comment]
Trying to reproduce this to get the full build log (actually in subsequent calls there was not really more output), I found out that something else must have fixed this. It must have been `sw/vendor/rustup/rustup-init.sh -y --default-toolchain 1.52.1` together with  `source $HOME/.cargo/env` what actually fixed it. I just changed two things and thought it was the other one that solved it.

Maybe this would be nice to have to include in the documentation that cargo needs to be in version <= 1.52 and the steps how to reach it.

Also I found out that GCC 10 must be used by prefixing `CC_FOR_BUILD=gcc-10 CXX_FOR_BUILD=g++-10 ./meson_init.sh -f` to the meson init.
